api/processors/actions/group/create.py

```diff
@@ -0,0 +1,40 @@
+from django.db import transaction
+
+from api.models.auxil.status_setter import StatusSetter
+from api.models.choices.log_event_type import (
+    GroupLogEventType,
+    StudentLogEventType,
+    TeacherLogEventType,
+)
+from api.models.choices.status import GroupStatus, StudentStatus, TeacherStatus
+from api.processors.actions.group import GroupActionProcessor
+from api.processors.auxil.log_event_creator import GroupLogEventCreator
+
+
+class GroupCreateProcessor(GroupActionProcessor):
+    @transaction.atomic
+    def process(self) -> None:
+        self._set_statuses()
+        self._create_log_events()
+
+    def _create_log_events(self) -> None:
+        GroupLogEventCreator.create(
+            group=self.group,
+            student_log_event_type=StudentLogEventType.GROUP_OFFERED,
+            teacher_log_event_type=TeacherLogEventType.GROUP_OFFERED,
+            group_log_event_type=GroupLogEventType.FORMED,
+        )
+
+    def _set_group_status(self) -> None:
+        StatusSetter.set_status(
+            obj=self.group, status=GroupStatus.PENDING, status_since=self.timestamp
+        )
+
+    def _set_coordinators_status(self) -> None:
+        return super()._set_coordinators_status()
+
+    def _set_teachers_status(self) -> None:
+        self.group.teachers.update(status=TeacherStatus.GROUP_OFFERED, status_since=self.timestamp)
```

Comment:
Here I am not sure about business logic , according to #123 this should be correct but according to @ivanBarinsky's code of group abort/start we actually need to update statuses of all teachers (not only ones assigned to this group). @lemontree210 What do you think?

Reply:
When we start/abort/finish the group, we actually _need_ to only update statuses of teachers (and, in the case of those actions, coordinators too) that were assigned to that group. But that would have led to a need to create more filters IIRC, so we discussed that with @Roxe322 and decided that we'd just update statuses of all teachers (and coordinators) because the statuses depend on number of groups and the result would be the same anyway. Since we use `.count()` to count groups for setting those statuses, no additional DB hits will be required anyway (unless I'm very wrong here).

Another thing that is special about this particular situation (with group being offered) is that, no matter whether the teacher is already working with another group or not, I want this situation (that the teacher was offered a group and either the bot or the coordinator are awaiting their reply ASAP) to be very clear and easily displayable to the coordinators.

The next step, then (this is not pertinent to this PR, just explaining the logic), will depend on whether the teacher and the sufficient number of students confirm their willingness to study: in case of success, we confirm the group and mark it as ready for start (#124) or discard it and reset the statuses for teachers according to how many groups they're still working with (#175).